engine/access/rpc/backend/backend_scripts.go

```diff
@@ -119,8 +104,8 @@ func (b *backendScripts) executeScriptOnExecutor(
 	// *DO NOT* use this hash for any protocol-related or cryptographic functions.
 	insecureScriptHash := md5.Sum(script) //nolint:gosec
 
-	// try execution on Archive nodes first
-	if len(b.archiveAddressList) > 0 {
+	// try execution on Archive nodes if there is no execution nodes found
+	if len(executors) == 0 && len(b.archiveAddressList) > 0 {
```

Comment:
I think we should keep the original. the intention of this code is to override the normal behavior and execute scripts on archive nodes instead. If there are not archive node configured or if the script fails, it should fall back to use execution nodes.